lib/customer.rb

```diff
@@ -0,0 +1,30 @@
+require 'csv'
+
+class Customer
+  attr_accessor :email, :address
+  attr_reader :id
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    all_customers = CSV.open("data/customers.csv", "r").map do |line|
+      Customer.new(line[0].to_i, line[1], "#{line[2]} #{line[3]} #{line[4]} #{line[5]}")
+    end
+    return all_customers
+  end
+
+  def self.find(id)
+    all_customers = self.all
+
+    all_customers.length.times do |i|
```

Comment:
Here, you're iterating over a collection/array of things. Instead of using `.length.times` and using `i` to index into the `all_customers` array, why not use a `.each` or an `Enumerable` method?